cfe_internal/enterprise/ha/ha_update.cf

```diff
@@ -0,0 +1,129 @@
+bundle agent ha_update
+{
+ vars:
+      "master_hub_location" string => "/var/cfengine/state/master_hub.dat";
+      "hubs_keys_location" string => "/var/cfengine/ppkeys_hubs"; 
+      "ppkeys_staging" string => "/var/cfengine/ppkeys_staging";
+ methods:
+  enable_cfengine_enterprise_hub_ha.am_policy_hub::
+      "share_keys" usebundle => ha_share_hub_keys;
+      "sync_hub_data" usebundle => ha_hub_sync;
+
+  enable_cfengine_enterprise_hub_ha::
+      "sync_agent_data" usebundle => ha_agent_sync;
+
+  enable_cfengine_enterprise_hub_ha.hub_data_synced::
+      "manage_keys" usebundle => manage_hub_synced_data;
+}
+
+bundle agent ha_agent_sync
+{
+ files:
+  !am_policy_hub::
+    "$(sys.workdir)/policy_server.dat"
+      copy_from => ha_no_backup_scp("$(ha_update.master_hub_location)", @(update_def.policy_servers)),
+      comment => "Update master hub IP on CFEngine node",
+      handle => "ha_cfengine_node_update_master";
+
+    "$(sys.workdir)/ppkeys"
+      copy_from => ha_no_backup_scp("$(ha_update.hubs_keys_location)", @(update_def.policy_servers)),
+      file_select => hub_all_keys,
+      depth_search => ha_recurse("inf"),
+      comment => "Distribute all hub keys to clients",
+      handle => "ha_copy_hub_keys_to_nodes";
+}
+
+bundle agent ha_share_hub_keys
+{
+ files:
+   "$(ha_def.ppkeys_hubs)"
+      copy_from => no_backup_cp("$(sys.workdir)/ppkeys"),
+      file_select => hubs_keys_select,
+      comment => "Copy all hubs keys to directory accessible by clients",
```

Comment:
Perhaps add something like "clients need to be able to download other hubs keys so that in the event of a fail over the client will be able to trust alternative hubs." to the comment?